common/ReduxChannel.js

```diff
@@ -0,0 +1,87 @@
+const DEFAULT_OPTIONS = {
+  timeout: 20000,
+  target: typeof window !== "undefined" ? window : {}
+};
+
+class Channel {
+  constructor(options = {}) {
+    if (!options.incoming) {
+      throw new Error("You must specify an incoming event name with the 'incoming' option.");
+    }
+    if (!options.outgoing) {
+      throw new Error("You must specify an outgoing event name with the 'outgoing' option.");
+    }
+    this.options = Object.assign({}, DEFAULT_OPTIONS, options);
+    this.callbacks = new Set();
+    this.timeouts = new Map();
+
+    this._onReceiveEvent = this.runCallbacks.bind(this);
+    this.options.target.addEventListener(this.options.incoming, this._onReceiveEvent);
```

Comment:
do you mind if I just do that in a follow-up ticket since this is just a straight up copy